tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlobStoreBackedInputStreamImpl.java

```diff
@@ -137,6 +137,18 @@ public void seekForward(long position) throws IOException {
         }
     }
 
+    @Override
+    public boolean haveDataToRead() {
+        log.debug("haveDataToRead cursor: {}, objectLen: {}, readableBytes: {}.",
+                cursor, objectLen, this.buffer.readableBytes());
```

Comment:
Nit: what do you think about wrapping this debug statement in a conditional to prevent unnecessary boxing as well as unnecessary computation?

Reply:
There is no extra method in `log.debug(xxx)`, the method `log.debug` will check the log level first and it seems that there is no extra cost.

Reply:
`this.buffer.readableBytes()` is the method I was referring to. It will be computed before `log.debug` is called. Also, before `log.debug` is called, the `cursor` and `objectLen` variables, which are both `long`, will be converted into `Long` causing a heap allocation. Since this `readAsync` method is on a critical path for reading data from tiered storage, I think these minor optimizations are valuable.

This was also discussed here: https://github.com/apache/pulsar/pull/11880#discussion_r700384425.

Reply:
in Pulsar we have this convention to always add log guards.
please add if (log.isDebugEnable()).

Reply:
Ok, I'll fix this.